### DISCUSSION TITLE
managing checkpoints: virtnbdls, virtnbdrm

### DISCUSSION THREAD

**Discussion by gitthangbaby** [abbbi/virtnbdbackup#156] [Ideas]
Thanks for the tool.

The tool lacks checkpoint management. It updates checkpoint into image, and updates its own files. But then, it leaves it as it is.
It provides no information about what is written on either side at the moment.
Since "auto" function cannot rotate target folder based on age or count of checkpoints, the use has to write the cleanup scripts. I was also surprised "full" mode is not able to delete the old checkpoints.

The result is exceptions and errors, which were added to the Issues several times:
checkpoint inconsistent: missing or broken bitmap 'virtnbdbackup.1' for disk 'sdc'

It needs to be constantly repaired, but that's double work, as also virtnbdbackup files need to be repaired, and given amount the files there, it's script intensive and bash scripts might outgrow the tool code. This logic would fit this high level tool much better.

So my idea is to have the tool be able to know what consistent checkpoints exist at the moment (on both sides), and be able to delete one of all of them.

**Comment by abbbi** [comment]
maybe you want to check on this project of mine:

https://github.com/abbbi/vircpt

but the situation you are describing really looks like you want to use the persistent checkpoint feature as described in the manual??
Are you migrating your virtual machines between hosts between backups? Please search the past issues and discussions on this topic.

If you run into situaions where the qcow image is missing an bitmap and libvirt still thinks here is an checkpoint, it *allways* needs manual intervention.

Please read:

https://github.com/abbbi/virtnbdbackup?tab=readme-ov-file#transient-virtual-machines-checkpoint-persistency

It should be clarified as of *why* you run in to this problem so frequently, its something in your workflow that is broken.
